### ISSUE TITLE
Provide an optional Wake On LAN (WOL) feature, per entry

### ISSUE THREAD

**Issue by kdorff** [louislam/uptime-kuma#810]
**Describe the solution you'd like**
It would be nice is there was an option, per entry, to send a Wake On LAN (WOL) packet when a service was observed as down. And likely an IP address to send the WOL packet to. Perhaps a configuration "don't send more often than..."? I'm not sure what options are generally provided around sending WOL.

**Describe alternatives you've considered**
Noticing the server is down (being notified by Kuma) and turning it back on manually.

**Comment by kdorff** [comment]
> This sounds like it could be implemented by having a notification type of Wake On Lan that sends the packet.

This sounds like a perfect, easy to understand solution.

**Comment by e-p-s** [comment]
+1

**Comment by Computroniks** [comment]
@kdorff Could you provide some information about a possible usecase for this feature. Would it be used to "Wake up" a server that has entered the low power state and has been detected down? I am not really sure of its usefulness as if a server is down, it is most likely because of a network error which wouldn't be helped by a WOL packet or the server has crashed (not sure if WOL would work here).

**Comment by Meulator** [comment]
@Computroniks I dont think it would be used with servers, but maybe with computers ; i think it would be useful, for people doing RDP to their fixed box from home for instance, having the opportunity to start it in case of power outage for instance.

**Comment by Meulator** [comment]
@SoulEater45 yes, you're right, but we are in a very specific configuration ; putting too much PC on restart when power back might create another outage due to the surge on power-on ; anyways, we found a way to do so with a single php webpage so we're good :)

**Comment by tomeurp** [comment]
This would be very useful indeed. I run a sql server at my business which is shut down every night to save on power consumption. This is fine because it is normally only needed during business hours. Ocasionally, however, I need to access the server via RDP outside of business hours and it would be pretty neat to be able to just wake the server from the wonderful interface of uptime kuma. I actually acomplish this via a user-defined-script in my synology nas, but it is a bit cumbersome and not user friendly at all to have to log into dsm open the control panel and run the script.
Hope this gets implemented soon.
Thanks.

**Comment by Head** [comment]
Usecase: Running a local DNS Server. DNS Check fails: WoL the backup server (which is powered off to save energy).

**Comment by ke1pl** [comment]
> [@Meulator](https://github.com/Meulator) that is true too, haven't thought about it! As for the WOL problem, I have also found a nice substitute called [upsnap](https://github.com/seriousm4x/UpSnap). It looks nice and has a network scan feature (which would be a nice addition to kuma, to make machine monitoring a bit easier/more convenient).

I have the same problem, the same solution and the same feature request that was described in this comment :)

**Comment by jztreso** [comment]
**Could you provide some information about a possible usecase for this feature.**
I for example run a home lab with multiple PCs. Some of them support powerup at power back, but some don't.
If my main NAS (which is running uptime kuma) boots up from a power outage, it can assist with booting up my other machines if it detects they aren't pinging back.
It could also be used to boot up VMs or machines at specific times a day.
If you seat the ping interval to just once per day, say at 8am, you could use it to boot up one, or a cluster at that time.

**Comment by tvup** [comment]
Different angle on this problem: instead of adding WoL to the monitoring tool, what about having the *proxy itself* wake the server on first request?

I built [idle-less](https://github.com/tvup/idle-less) — a Docker reverse proxy that detects when a backend is down and sends a WoL magic packet automatically. Visitors see a waiting screen while the server boots, then get redirected.

It's complementary to Uptime Kuma — idle-less handles the "wake on first HTTP request" pattern, while Uptime Kuma monitors availability. Together they cover both use cases.

- Docker Compose, runs on Pi (ARM64)
- Multi-domain → multi-server WoL
- macvlan for L2 broadcast

GitHub: https://github.com/tvup/idle-less
Demo: https://tvup.github.io/idle-less/demo.html
